### PR TITLE
Add the support class for the Adjoint Jacobian to the new device

### PR DESCRIPTION
**Pull request by LuisAlfredoNu**
### Before submitting

Please complete the following checklist when submitting a PR:

- [X] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [X] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [X] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [X] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
 Migrate LightningGPU to the new device API

**Description of the Change:**
Create the `adjoint-jacobian` class for the new device API to achieve the `jacobian` and `vjp` methods

**Benefits:**
Unlocking the `adjoint-jacobian` capabilities with the new device API for LGPU

**Possible Drawbacks:**

**Related GitHub Issues:**
## **Freezzed PR** :warning: :snowflake: 
To make a smooth integration of LightningGPU with the new device API, we set the branch `gpuNewAPI_backend` as the base branch target for future developments related to this big task. 

The branch `gpuNewAPI_backend` has the mock of all classes and methods necessary for the new API. Also, several tests were disabled with 
``` python
if device_name == "lightning.gpu":
    pytest.skip("LGPU new API in WIP.  Skipping.",allow_module_level=True)
```
However, these tests will unblocked as the implementation progresses.

After all the developments for integrating LightningGPU with the new API have been completed then the PR will be open to merge to `master`

[sc-70936] [sc-70939]

### CODE REVIEW

**Comment by AmintorDusko** [comment]
Hey @LuisAlfredoNu, is this PR ready for review? Apparently, we have several test fails.

**Comment by LuisAlfredoNu** [comment]
> Hey @LuisAlfredoNu, is this PR ready for review? Apparently, we have several test fails.

@AmintorDusko 
Sorry, I realize that right now it has several errors after the merge of the last commit of `simulate` branch :disappointed:. I will fix it and call to review it again. :smile:

**Comment by AmintorDusko** [comment]
> > Hey @LuisAlfredoNu, is this PR ready for review? Apparently, we have several test fails.
> 
> @AmintorDusko Sorry, I realize that right now it has several errors after the merge of the last commit of `simulate` branch 😞. I will fix it and call to review it again. 😄

Thank you, @LuisAlfredoNu!

File: pennylane_lightning/lightning_kokkos/_state_vector.py [hunk 1]
```diff
@@ -62,15 +62,13 @@ def __init__(
...
         kokkos_args=None,
-        sync=True,
     ):  # pylint: disable=too-many-arguments
 
         super().__init__(num_wires, dtype)
 
         self._device_name = "lightning.kokkos"
 
         self._kokkos_config = {}
-        self._sync = sync
```
Comment (LuisAlfredoNu) [inline pennylane_lightning/lightning_kokkos/_state_vector.py:73]: In LK, any method requires the var `sync`

File: pennylane_lightning/lightning_gpu/_adjoint_jacobian.py [hunk 2]
```diff
@@ -15,9 +15,37 @@
...
+        warn(str(ex), UserWarning)
+        MPI_SUPPORT = False
+
+except ImportError as ex:
+    warn(str(ex), UserWarning)
+    pass
+
 import numpy as np
-import pennylane as qml
+from pennylane.tape import QuantumTape
```
Comment (maliasadi) [inline pennylane_lightning/lightning_gpu/_adjoint_jacobian.py:46]: :+1:

**Review by maliasadi — APPROVED** [review]
Nice work! Thanks @LuisAlfredoNu! Happy to approve :100:

**Review by AmintorDusko — APPROVED** [review]
Great job!

**Pull request merged**